### ISSUE TITLE
ERROR: Assign requires shapes of both tensors to match. lhs shape= [1000] rhs shape= [1001]

### ISSUE THREAD

**Issue by apiszcz** [tensorflow/tensorflow#6883]
Searched for possible solutions, this page references using --Label-offsets=1 to use vgg or ResNet, here however I'm using InceptionV3 with imagenet TFRecord format, and the failure message occurs during initialization of the session.

Operating System: Windows 10/64 or LINUX

Installed version of CUDA and cuDNN: 
cublas64_80.dll locally
cudnn64_5.dll
cufft64_80.dll
nvcuda.dll locally
curand64_80.dll locally

If installed from binary pip package, provide:
0.12.1

CODE:
    init_fn = slim.assign_from_checkpoint_fn(model_path,slim.get_model_variables('InceptionV3'))

    with tf.Session() as sess:
        # Load weights
        init_fn(sess)

ERROR MESSAGE
InvalidArgumentError (see above for traceback): Assign requires shapes of both tensors to match. lhs shape= [1000] rhs shape= [1001]
         [[Node: save/Assign_8 = Assign[T=DT_FLOAT, _class=["loc:@InceptionV3/AuxLogits/Conv2d_2b_1x1/biases"], use_locking=true, validate_shape=true, _device="/job:localhost/replica:0/task:0/cpu:0"](InceptionV3/AuxLogits/Conv2d_2b_1x1/biases, save/RestoreV2_8)]]

**Comment by yaroslavvb** [comment]
That doesn't seem like a bug in tensorflow -- that errors means you are a loading 1001 elements into variable with 1000 elements

**Comment by apiszcz** [comment]
Thanks. W

**Comment by weizier** [comment]
If you have a directory including the model trained before, you could try to remove it, then re-run.

**Comment by luyishisi** [comment]
work！

**Comment by warisha** [comment]
thanks :+1:  it works!

**Comment by aforslow** [comment]
In my case, I got the error when distributing my training session into multiple worker threads. What solved it for me was surrounding the inference graph with a tf.variable_scope(name, reuse=tf.AUTO_REUSE). I guess tensorflow can't keep track of your variables when working in a distributed mode.

**Comment by dshahrokhian** [comment]
It is absolutely insane that the error message has no relation what so ever with the solution. Any insights on why does this happen?

**Comment by YinniCoding** [comment]
> If you have a directory including the model trained before, you could try to remove it, then re-run.

If I delete them, my program how to load the trained model  ???

**Comment by ChChwang** [comment]
> > If you have a directory including the model trained before, you could try to remove it, then re-run.
> 
> If I delete them, how can my program load the trained model ???

finetune?

**Comment by Jemmagu** [comment]
> > > If you have a directory including the model trained before, you could try to remove it, then re-run.
> > 
> > 
> > If I delete them, how can my program load the trained model ???
> 
> finetune?

yes, fine-tune Inception V3 model. what does it mean by removing pertained model?

**Comment by YinRui1991** [comment]
Hi, @apiszcz You should look the function "assign_from_checkpoint_fn ( ) ". Find code 
"for var in var_dict:", the below codes will load each layer weights. You can print "var" to see all the layer names, and wirte codes to let the sess not load the layer which has different shape.

**Comment by Deepcong2019** [comment]
> If you have a directory including the model trained before, you could try to remove it, then re-run.

why do this, what the trained model do???

**Comment by mikechen66** [comment]
ImageNet defaults 1000 classes but the related weights downloaded has 1001 classes. It it the source of the conflict. 

It is necessary to set "labels_offset=1". I know the following solution in the bazel environment. But I have not know where "labels_offset=1" shall be inserted in the non-bazel environment.

Notes:

For the bazel environment, please take the following reference. 

./bazel-bin/slim/train
--train_dir=${TRAIN_DIR}
--dataset_dir=${DATASET_DIR}
--dataset_name=imagenet
--dataset_split_name=train
--model_name=resnet_v1_50
--checkpoint_path=${CHECKPOINT_PATH}
--labels_offset=1

**Comment by SayaliNagwkar17** [comment]
> > > > > If you have a directory including the model trained before, you could try to remove it, then re-run.
> > > > 
> > > > 
> > > > If I delete them, how can my program load the trained model ???
> > > 
> > > 
> > > finetune?
> > 
> > 
> > yes, fine-tune Inception V3 model. what does it mean by removing pertained model?
> 
> if finetune, don's remove the pretrained model and get part of parameters

Did you get the solution? I also want to fine tune my model
